src/components/Leaderboard.tsx

```diff
@@ -0,0 +1,79 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import {
+    Table,
+    Thead,
+    Tbody,
+    Tr,
+    Th,
+    Td,
+    TableCaption,
+    TableContainer,
+    Container,
+    Heading,
+    Button,
+} from "@chakra-ui/react";
+import { Podium } from "./Podium";
+
+export interface leaderboard {
+    id: number;
+    breedname: string;
+    votes: number;
+}
+export function Leaderboard(): JSX.Element {
+    const backend = "https://dog-breed-voting-c7b1.onrender.com";
+
+    const [leaderboard, setLeaderboard] = useState<leaderboard[]>([]);
+
+    const getLeaderboard = async () => {
+        const response = await axios.get(`${backend}/leaderboard`);
+        setLeaderboard(response.data);
+    };
+    useEffect(() => {
+        getLeaderboard();
+    }, []);
+
+    const leaderboardCopy = [...leaderboard];
+    const top3 = leaderboardCopy.splice(0, 3);
+
+    const leaderboardTableRows = leaderboardCopy.map((row, index) => (
+        <Tr key={index}>
+            <Td>{index + 4}</Td>
+            <Td>{row.breedname}</Td>
+            <Td>{row.votes}</Td>
+        </Tr>
+    ));
+
+    return (
+        <>
+            <Container marginTop={"2vh"} centerContent>
+                <Heading marginTop={"2vh"} textAlign={"center"}>
+                    Current Leaderboard:
+                </Heading>
+                <Button
+                    onClick={() => {
+                        getLeaderboard();
+                    }}
+                    colorScheme="teal"
+                    margin={"2vh"}
+                >
+                    Refresh Leaderboard
+                </Button>
+                {top3.length > 0 && <Podium top3Array={top3} />}
```

Comment:
Should we check that the top3.length >= 3 instead of zero. So that we display the podium when we have at least three breeds voted for.